swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift

```diff
@@ -12,6 +12,8 @@ import SwiftUI
 @MainActor
 public final class SessionViewModel: ObservableObject {
   @Published private(set) var actorName: String? = nil
+  /// Should be nil until the Portal tells us the Resources list
+  /// Nil is distinct from an empty list
   @Published private(set) var resources: [Resource]? = nil
```

Comment:
I know barely any Swift but if it is easy, could we model this in the type system with an enum or something? Such subtle behaviour based on `nil` is bound to be overlooked again and comments aren't yet causing compile errors :)

In Rust, I'd probably make an enum that implements `IntoIterator` (if that is the contract that the view needs to list display the resources.

Reply:
Yeah this will happen again won't it. It should really be something like
```rust
enum ResourcesList {
    Loading,
    Loaded(Vec<Resource>),
}
```

ChatGPT says...

```swift
enum ResourcesList {
    case loading
    case loaded([Resource])
}
```

So basically an `Option<Vec<Resource>>` but without the semantics of `Option` to confuse everything.